### PR TITLE
docs: add networking section (close #4346)

### LINKED ISSUE #4346: add networking section to docs
Add a networking section to the docs.

Tables with the following dimensions:
- OS: Mac/windows/linux
- External API (in docker or outside docker)
- URL for Hasura to API connectivity
- URL for API to Hasura connectivity

### PR DESCRIPTION
**Pull request by marionschleifer**
### Description

This PR adds a networking section to the deployment part in docs.

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [x] Docs

### Related Issues

https://github.com/hasura/graphql-engine/issues/4346

### Steps to test and verify

https://deploy-preview-4811--hasura-docs.netlify.app/graphql/manual/guides/docker-networking.html

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -69,6 +69,7 @@ Read more about the session argument for computed fields in the [docs](https://h
 - docs: add single object mutations docs (close #4622) (#4625)
 - docs: add docs page on query performance (close #2316) (#3693)
 - docs: add a sample Caddyfile for Caddy 2 in enable-https section (#4710)
+- docs: add docs page on network config (close #4346) (#4811)
```
Comment (tirumaraiselvan) [inline CHANGELOG.md:72]:
```suggestion
- docs: add docs page on networking with docker (close #4346) (#4811)
```

**Comment by marionschleifer** [comment]
@rikinsk I agree that the positioning is a little weird but didn't have a better idea then. Let's make it its own section under Guides then? I can change that and also link there from all the pages where the networking stuff is relevant.

File: docs/graphql/manual/guides/docker-networking.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+.. contents:: Table of contents
+  :backlinks: none
+  :depth: 2
+  :local:
+
+Introduction
+------------
+
+Sometimes you might want to connect Hasura with APIs that are either running outside of Docker or in a different Docker container.
```
Comment (tirumaraiselvan) [inline docs/graphql/manual/guides/docker-networking.rst:18]: can we give examples of APIs here (auth webhook, event triggers, remote schemas)

File: docs/graphql/manual/guides/docker-networking.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+   * - 
+     - Mac
+     - Windows
+     - Linux
+     - Comment
+   * - **Hasura to API (outside Docker)**
+     - ``host.docker.internal:3000``
+     - ``docker.for.win.localhost:3000``
+     - ``localhost:3000`` with (``--net=host``)
```
Comment (tirumaraiselvan) [inline docs/graphql/manual/guides/docker-networking.rst:36]: We need to account for the case without `--net=host` as that will be more common esp if Hasura was launched via docker-compose.

E.g. `172.17.0.1:3000` (see docker bridge ip via `ifconfig`)

File: docs/graphql/manual/guides/docker-networking.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+     - service name, e.g.: ``api:3000``
+     - service name, e.g.: ``api:3000``
+     - service name, e.g.: ``api:3000``
+     - Assuming the API is running on port ``3000``
+   * - **API to Hasura (both in docker-compose)** 
+     - service name, e.g.: ``hasura:8080``
+     - service name, e.g.: ``hasura:8080``
+     - service name, e.g.: ``hasura:8080``
+     - Hasura runs on port ``8080`` by default
+   * - **Hasura to API (both running with docker run)**
```
Comment (tirumaraiselvan) [inline docs/graphql/manual/guides/docker-networking.rst:53]: Can we say `(both running with separate docker run)`

File: CHANGELOG.md [hunk 5]
```diff
@@ -17,6 +17,7 @@
 - server: process events generated by the event triggers asynchronously (close #5189) (#5352)
 - console: display line number that error originated from in GraphQL editor (close #4849) (#4942)
 - docs: add page on created_at / updated_at timestamps (close #2880) (#5223)
+- docs: add docs page on networking with docker (close #4346)
```
Comment (tirumaraiselvan) [inline CHANGELOG.md:20]: this needs to come up I think

**Pull request merged**